### PR TITLE
add getCollapsedText(query) to DataSourceApi

### PR DESCRIPTION
**Pull request by ryantxu**
Part of #16481 and #16059

Currently, only angular components can set the `getCollapsedText`.  This PR adds `getCollapsedText` to DataSourceApi and uses it if it exists

### CODE REVIEW

**Review by dprokop — APPROVED** [review]
LGTM. Not sure about `getCollapsedText` name tho. WDYT about `getQueryText` or  `getRawQuery`?

File: packages/grafana-ui/src/types/datasource.ts [hunk 1]
```diff
@@ -106,6 +106,11 @@ export interface DataSourceApi<TQuery extends DataQuery = DataQuery> {
    */
   getQueryHints?(query: TQuery, results: any[], ...rest: any): QueryHint[];
 
+  /**
+   * Convert a query to a simple text string
+   */
+  getCollapsedText?(query: TQuery): string;
```
Comment (torkelo) [inline packages/grafana-ui/src/types/datasource.ts]: * getCollapsedQueryText(query)    feels more specific

**Review by torkelo — COMMENTED** [review]
nice!

File: packages/grafana-ui/src/types/datasource.ts (hunk 1 above)
Comment (ryantxu) [reply packages/grafana-ui/src/types/datasource.ts]: I'll change it -- I just copied from the angular version :)

Comment (dprokop) [reply packages/grafana-ui/src/types/datasource.ts]: What is the difference between query's text and collapsed text? Not sure if the `collapsed` adjective is not blurring the essence of this method.

Comment (torkelo) [reply packages/grafana-ui/src/types/datasource.ts]: Well it's a shortform, "display" form, might not show everything, only essentials. For data sources that have non string based query languages like elasticsearch  it just becomes more of a render essential options to a formatted string.

Comment (dprokop) [reply packages/grafana-ui/src/types/datasource.ts]: OK, then it makes sense. Since it was confusing I suggest adding an example in the comment above.

Comment (dprokop) [reply packages/grafana-ui/src/types/datasource.ts]: I like `getQueryDisplayText`

Comment (ryantxu) [reply packages/grafana-ui/src/types/datasource.ts]: done, and passed CI.  

Still something funny with my permissions that I can't merge things after they are approved.  This is what I see:
image

**Pull request merged**